pkg/api/dirs.go

```diff
@@ -169,8 +168,8 @@ func storeDir(ctx context.Context, reader io.ReadCloser, s storage.Storer, mode
 		return swarm.ZeroAddress, fmt.Errorf("metadata marshal: %w", err)
 	}
 
-	sp := splitter.NewSimpleSplitter(s, mode)
-	mr, err := file.SplitWriteAll(ctx, sp, bytes.NewReader(metadataBytes), int64(len(metadataBytes)), toEncrypt)
+	pipe := pipeline.NewPipeline(ctx, s, mode)
+	mr, err := pipeline.FeedPipeline(ctx, pipe, bytes.NewReader(metadataBytes), int64(len(metadataBytes)))
```

Comment:
Maybe you can inline pipe here?

Reply:
let's leave as it is as things might get more complicated when encryption is introduced